perfkitbenchmarker/benchmarks/scimark2_benchmark.py

```diff
@@ -0,0 +1,265 @@
+        if clear:
+          result = result[:m.start()] + ' ' * (m.end() - m.start()) + \
+              result[m.end():]
+    return metadata, result
+
+  def ExtractScore(result, bench_version, clear=True):
+    m = re.search(SCORE_REGEX, result)
+    label = m.group(1)
+    score = float(m.group(2))
+    if clear:
+      result = result[:m.start()] + ' ' * (m.end() - m.start()) + \
+          result[m.end():]
+    return score, label, result
+
+  def ExtractResults(result, bench_version):
+    datapoints = []
+    if bench_version == 'C':
+      match = regex_util.ExtractAllMatches(RESULT_REGEX_C, result)
+      for groups in match:
+        metric = '{0} {1}'.format(groups[0].strip(), groups[2].strip())
+        metric = metric.strip().strip(':')
+        value = float(groups[1])
+        datapoints.append((metric, value))
+    elif bench_version == 'Java':
+      match = regex_util.ExtractAllMatches(RESULT_REGEX_JAVA, result)
+      for groups in match:
+        datapoints.append((groups[0].strip(), float(groups[1])))
+    return datapoints
+
+  samples = []
+  start_index, bench_version = FindBenchStart(results)
+  while start_index != -1:
+    next_start, next_bench = FindBenchStart(results, start_index + 1)
+    result = results[start_index:next_start]
+
+    metadata = {'bench_version': bench_version}
+
+    platform_metadata, result = ExtractPlatform(result, bench_version)
+    metadata.update(platform_metadata)
+
+    score, label, result = ExtractScore(result, bench_version)
+    samples.append(sample.Sample(label, score, 'Mflops', metadata))
+
+    datapoints = ExtractResults(result, bench_version)
+    for metric, value in datapoints:
+      samples.append(sample.Sample(metric, value, 'Mflops', metadata))
+
+    start_index, bench_version = next_start, next_bench
+
+  return samples
+
+
+def TestParseResults():
+  TEST_C = """**                                                              **
+** SciMark2 Numeric Benchmark, see http://math.nist.gov/scimark **
+** for details. (Results can be submitted to pozo@nist.gov)     **
+**                                                              **
+Using       2.00 seconds min time per kenel.
+Composite Score:         1596.04
+FFT             Mflops:  1568.64    (N=1024)
+SOR             Mflops:  1039.98    (100 x 100)
+MonteCarlo:     Mflops:   497.64
+Sparse matmult  Mflops:  1974.39    (N=1000, nz=5000)
+LU              Mflops:  2899.56    (M=100, N=100)
+"""
+  TEST_JAVA = """SciMark 2.0a
+
+Composite Score: 1716.3662351463677
+FFT (1024): 1000.1380057152871
+SOR (100x100):   1353.1987180103354
+Monte Carlo : 727.7138820888014
+Sparse matmult (N=1000, nz=5000): 1495.40225150659
+LU (100x100): 4005.3783184108247
+
+java.vendor: Oracle Corporation
+java.version: 1.7.0_75
+os.arch: amd64
+os.name: Linux
+os.version: 3.16.0-25-generic
+"""
+  samples = []
+  samples.extend(ParseResults(TEST_C))
+  samples.extend(ParseResults(TEST_JAVA))
+  return samples
```

Comment:
once moved to the tests directory it will be good to make sure the values are correct.